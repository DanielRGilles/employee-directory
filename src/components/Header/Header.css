
.headerT {
    display: grid;
    grid-template-columns: 15% 1fr 15%;
    background-color: rgb(121, 68, 68);
    color: cornsilk;
  }
  .img {
      width: 45px;
  }