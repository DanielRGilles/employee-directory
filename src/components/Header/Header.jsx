import React from 'react'
import { Link } from 'react-router-dom';
import { useUser } from '../../context/UserContext';
import { signOutUser } from '../../services/users';
import styles from './Header.css'
import phone  from '../../assets/gymnast.svg';

export default function Header() {
    const { user, setUser } = useUser();
    
   
    return (
        <header className={styles.headerT}>
            <div></div>
            <div>
            <h1>Acme Inc. </h1>
            <div><img className={styles.img}  src={phone} alt='karate kicker'/></div>
            <div>{user?.email ? user.email : 'Friend'}</div></div>
            <div>
            {user?.email ? <button onClick={async () => { await signOutUser();
            setUser({})}} className='auth-btn'>Sign Out</button> : (
                <Link to="/login" className='auth-btn'>
                  <button>Sign In</button>
                </Link>
              )}</div>
        </header>
    )
}
