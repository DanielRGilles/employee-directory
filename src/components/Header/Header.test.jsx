import {render } from  '@testing-library/react';
import { MemoryRouter } from 'react-router-dom/cjs/react-router-dom.min';
import { UserProvider } from '../../context/UserContext';
import Header from './Header';

jest.mock('../../context/UserContext')
jest.mock('../../services/users.js')

describe('Header component test', () => {
  
    it('renders the top nav header', () => {
       const { container } = render(
            <MemoryRouter>
                <UserProvider>
                     <Header/>
                </UserProvider>
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
         
    })

})