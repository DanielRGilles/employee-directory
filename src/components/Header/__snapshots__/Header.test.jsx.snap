// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test renders the top nav header 1`] = `
<div>
  <header
    class="headerT"
  >
    <div />
    <div>
      <h1>
        Acme Inc. 
      </h1>
      <div>
        <img
          alt="karate kicker"
          class="img"
          src="/path/image"
        />
      </div>
      <div>
        Friend
      </div>
    </div>
    <div>
      <a
        class="auth-btn"
        href="/login"
      >
        <button>
          Sign In
        </button>
      </a>
    </div>
  </header>
</div>
`;
